Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift

```diff
@@ -106,12 +115,20 @@ public final class DefaultSettingsProvider: DefaultSettingsProviding {
 
     // MARK: - Private
 
-    private func createFilter(defaultSettings: DefaultSettings, essentialKeys: Set<String>) -> (String, SettingValue) -> Bool {
+    private func createFilter(defaultSettings: DefaultSettings,
+                              essentialKeys: Set<String>,
+                              newXcodeKeys: [Version: Set<String>] = [:]) throws -> (String, SettingValue) -> Bool {
+        let xcodeVersion = try XcodeController.shared.selectedVersion()
```

Comment:
I think caching is the most elegant solution, rather than passing the value around, will add it.

Speaking of performance, maybe it'd be beneficial to add some kind of speed tests to catch these performance regressions automatically? For example [swiftlint](https://github.com/realm/SwiftLint) runs those for a couple of open source real-world examples; now that's probably not feasible for us at the moment (not enough big OSS projects using tuist), but at least creating a dummy project would suffice, I think.
The output could then be quite similar to [this](https://github.com/realm/SwiftLint/pull/2987#issuecomment-563983472). 
What do you think?

Reply:
I have seen the [proposal](https://github.com/tuist/tuist/issues/820) now, nevermind, you were faster 😅

Reply:
Added the caching, let me know if that's what you envisioned.

Reply:
Thanks! - one minor tweak is moving the version lookup to the `.recommended:` case - that way we don't incur an io operation at all for `essential` case.  

```swift
case .recommended:
   let xcodeVersion = try XcodeController.shared.selectedVersion()
   return { key, _ in 
   // ...
```